tests/tests/trace_api.rs

```diff
@@ -188,7 +193,83 @@ async fn test_debug_trace_block_by_number(#[future] plain_opcodes: (Katana, Kaka
         serde_json::from_value(raw["result"].clone()).expect("Failed to deserialize result");
 
     assert!(traces.is_some());
-    // We expect 3 traces per transaction: CALL, CREATE, and CALL.
+    // We expect 1 trace per transaction given the formatting of the debug_traceBlockByNumber response.
     assert!(traces.unwrap().len() == TRANSACTIONS_COUNT);
     drop(server_handle);
 }
+
+#[rstest]
+#[awt]
+#[tokio::test(flavor = "multi_thread")]
+async fn test_trace_eip3074(#[future] eip_3074_invoker: (Katana, KakarotEvmContract, KakarotEvmContract), _setup: ()) {
+    // Setup the Kakarot RPC server.
+    let katana = eip_3074_invoker.0;
+    let counter = eip_3074_invoker.1;
+    let invoker = eip_3074_invoker.2;
+
+    let eoa = katana.eoa();
+    let eoa_address = eoa.evm_address().expect("Failed to get eoa address");
+
+    let chain_id = katana.eth_provider().chain_id().await.expect("Failed to get chain id").unwrap_or_default().to();
+    let invoker_address = Address::from_slice(&invoker.evm_address.to_bytes_be()[12..]);
+    let commit = B256::default();
+
+    let get_args = Box::new(move |nonce| {
+        // Taken from https://github.com/paradigmxyz/alphanet/blob/87be409101bab9c8977b0e74edfb25334511a74c/crates/instructions/src/eip3074.rs#L129
+        // Composes the message expected by the AUTH instruction in this format:
+        // `keccak256(MAGIC || chainId || nonce || invokerAddress || commit)`
+        fn compose_msg(chain_id: u64, nonce: u64, invoker_address: Address, commit: B256) -> B256 {
+            let mut msg = [0u8; 129];
+            msg[0] = 0x4;
```

Comment:
```suggestion
            // MAGIC constant is used for [EIP-3074](https://eips.ethereum.org/EIPS/eip-3074) signatures to prevent signature collisions with other signing formats.
            msg[0] = 0x4;
```